.github/workflows/rustdoc.yml

```diff
@@ -0,0 +1,68 @@
+name: Deploy rustdoc
+
+# GitHub Action workflow documentation:
+# https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions
+
+# This workflow is based on
+# https://github.com/actions/starter-workflows/blob/main/pages/mdbook.yml
+
+on:
+  push:
+    branches: [$default-branch]
+    paths:
+      - src/**
+      - Cargo.toml
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
```

Comment:
if this is used here, it makes sense to also add those lines to the book deployment workflow